.buildkite/pipeline.yml

```diff
@@ -21,7 +21,7 @@ steps:
   - group: "Linters"
     steps:
       - label: ":swift: SwiftLint"
-        command: .buildkite/commands/run-swiftlint.sh
+        command: .buildkite/commands/run-swiftlint.sh --strict
```

Comment:
I know that the PR description says that this is intentional… but this will make BOTH errors and warnings be reported as errors (like the `TREAT_WARNINGS_AS_ERRORS` build setting in Xcode).

While I love the commitment of adhering to the zero-warning policy, I'm just wondering if that's something that devs have agreed on?

For example:
 - Some people find this too strict, especially if they're working on a feature iteratively, using multiple PRs all targeting the feature branch to shape the code one step at a time, and would want to allow warnings in the intermediate PR, and only get rid of the warnings and clean up things on the last steps, etc.
 - Other people prefer this strict mode to not let the tech dept slip.
 - A compromise could be to only enforce `--strict` if the PR is targeting `trunk` or `release/*`, but still allow PRs with swiftlint warnings to land if they're targeting an intermediate feature branch, for example 🤔

Reply:
> While I love the commitment of adhering to the zero-warning policy, I'm just wondering if that's something that devs have agreed on?

Given we're making changes to the SwiftLint setup and this is also how I'd personally prefer it set up, I used this PR also as a way to raise the discussion of a zero-warning policy with the team. Thanks for highlighting it so that's clearer!
I do understand that perhaps this isn't for all teams and that's why I also liked your suggestion:

> A compromise could be to only enforce --strict if the PR is targeting trunk or release/*, but still allow PRs with swiftlint warnings to land if they're targeting an intermediate feature branch, for example 

I can see some teams preferring it that way, that's a great suggestion!

Reply:
What about a RFC P2? Otherwise not sure the whole PCiOS team will see this PR and thus consider it as a kind of poll on the question… 🤔

Reply:
The RFC was posted at pdeCcb-4qy-p2